lib/LaTeXML/Core.pm

```diff
@@ -212,7 +212,7 @@ sub convertDocument {
           my @dedup = ();
           while (my $check = shift(@copy)) {
             unshift(@dedup, $check) if !(grep { $_ eq $check } @dedup); }
-          $document->insertPI('latexml', searchpaths => join(',', @dedup)); } }
```

Comment:
> lists of URLs: multiple URLs are now space separated, which is safe, since spaces within URLs will be percent-encoded

Could you share the motivation for dispencing with the comma? That was safe too I think.

There may be code in various places external to latexml expecting the current separator char (since they get serialized out), so this will be a backwards-incompatible change.

Reply:
> Could you share the motivation for dispencing with the comma? That was safe too I think.

Not if the file name contains a comma!

Reply:
Er, a URI-encoded comma is %2C right?

Reply:
> this will be a backwards-incompatible change.

True, this would have to be documented. On the bright side, using file URLs means that serialisation is a known quantity (e.g. libexslt implements `str:decode-uri` and `str:encode-uri`).

Reply:
> Er, a URI-encoded comma is %2C right?

Good point, but, I just checked and URI::file does not encode the comma (it's not a special character after all).

Reply:
Right, but I feel even less comfortable knowing that now. 

I'd expect either no attempt at encoding, or a complete URI-encoding for methods that claim `...to_url` (or similar). I've used [URI::Encode](https://metacpan.org/pod/URI::Encode) for that in the past.

I think the master version of the method you modified was mainly interested in using `/` separator chars for the paths and did no encoding. 

We certainly didn't support all OS-legal filenames, that's definitely a good critique. But I am actually aware of two projects I need to go and refactor from splitting on commas to splitting on space, and then also add a URI-decode step, which is painful. I wonder who else is relying on the current convention, and how quietly this change will lead to breakage for people...

Reply:
(Absolute paths will keep starting with file:// though.)

Reply:
I changed `pathname_to_url` so that the comma is also encoded, and the file:// part is omitted. That preserves the previous behaviour on Unix-like systems except for some special characters getting encoded. I believe the behaviour is correct also w.r.t. Win32 UNCs, but that's harder to test properly.